backend/dataall/modules/shares_base/db/share_object_models.py

```diff
@@ -57,3 +58,17 @@ class ShareObjectItem(Base):
     healthStatus = Column(String, nullable=True)
     healthMessage = Column(String, nullable=True)
     lastVerificationTime = Column(DateTime, nullable=True)
+    attachedDataFilterUri = Column(
+        String, ForeignKey('share_object_item_data_filter.attachedDataFilterUri'), nullable=True
+    )
+
+
```

Comment:
Just curious. The implementation of ShareObjectItemDataFilter surprised me a bit, my first impulse would have been to leave ShareObjectItem without any additional columns and implement ShareObjectItemDataFilter with a reference to ShareObjectItem (same way ShareObject-ShareObjectItem do)

```
class ShareObjectItemDataFilter(Base):
    __tablename__ = 'share_object_item_data_filter'
    attachedDataFilterUri = Column(String, primary_key=True, default=utils.uuid('shareitemdatafilter'))
    label = Column(String, nullable=False)
    ++ shareObjectItemUri = Column(String, nullable=False)
    ++ dataFilterUri = Column(String, nullable=False)
    ++ dataFilterName = Column(String, nullable=False)
    -- dataFilterUris = Column(ARRAY(String), nullable=False)
    -- dataFilterNames = Column(ARRAY(String), nullable=False)
    itemUri = Column(String, nullable=False)
```

But looking at the PR solution, the queries might be faster, we don't need to resolve as many subqueries. I am just curious to understand why hahahhah

Reply:
Honestly i just implemented how I thought made sense did not do much digging into other instances of the same but you are right it is a bit different

I do think it is helpful to quick determine if a share item has filters or not without having to do a separate query on `share_object_item_data_filter` to resolve if a data filter exists for that share item - what do you think?

Reply:
We would always work with dataset-specific filters, we will never do a full list or query filters from various datasets. That's why I think this implementation makes more sense in this particular case. I think I'll copy it for some cases with the same parent-child relationship